app/Article.php

```diff
@@ -146,16 +146,42 @@ public function getReadingTimeAttribute()
 
     public function getParsedContentAttribute()
     {
-        $content = Str::of(html_entity_decode($this->content))->replaceMatches("/\[article_teaser id=(.*?)\]/", function ($match) {
-            if (empty($match[1])) return '';
-            $id = Str::of($match[1]);
-            $article = Article::published()->find($id);
-
-            if(!$article) return '';
-            
-            $article_teaser = Blade::render('<x-article_teaser :article="$article" />', ['article' => $article]);
-            return $article_teaser;
-        });
+        $content = Str::of(html_entity_decode($this->content))
+            ->replaceMatches('/\[article_teaser id=(.*?)\]/', function ($match) {
+                if (empty($match[1])) {
+                    return '';
+                }
+                $id = Str::of($match[1]);
+                $article = Article::published()->find($id);
+
+                if (!$article) {
+                    return '';
+                }
+
+                $article_teaser = Blade::render('<x-article_teaser :article="$article" />', [
+                    'article' => $article,
+                ]);
+                return $article_teaser;
+            })
+            ->replaceMatches('/\[collection_teaser id=(.*?)\]/', function ($match) {
+                if (empty($match[1])) {
+                    return '';
+                }
+                $id = Str::of($match[1]);
+                $collection = Collection::published()->find($id);
+
+                if (!$collection) {
+                    return '';
+                }
+
+                $collection_teaser = Blade::render(
+                    '<x-collection_teaser :collection="$collection" />',
+                    [
+                        'collection' => $collection,
+                    ]
+                );
+                return $collection_teaser;
+            });
         return $content;
```

Comment:
I would really like all this parsing and rendering logic to be tucked in a dedicated component. 

Something like this:
```suggestion
        return Str::of(html_entity_decode($this->content))
            ->replaceMatches('/\[article_teaser id=(.*?)\]/', function ($match) {
               return Teaser::buildFromMarkup($match[0])::render()
            });
```

Then this component would have things like
```php
namespace App\View\Components;

use Illuminate\View\Component;

class Teaser extends Component
{
    private string $type;
    private int $id;

    /**
     * Create a new component instance.
     *
     * @return void
     */
    public function __construct() {
        // ...
    }

    public static function buildFromMarkup(string $markup): Teaser
    {
       //... all the parsing happens here
    }

    public function render()
    {
        // rendering and probably even data fetching happens here
        if ($this->type === 'article') {
          // render x-article_teaser ...
        }
        // render something else
        // or render empty string
    }
}
```